app/helpers/samples_helper.rb

```diff
@@ -63,7 +63,47 @@ def parsed_samples_for_s3_path(s3_path, project_id, host_genome_id)
         sample_list << sample_attributes
       end
     end
-
     sample_list
   end
+
+  def samples_output_data(samples)
+    final_result = []
+    samples.each do |output|
+      output_data = {}
+      pipeline_info = output.pipeline_runs.first ? output.pipeline_runs.first.pipeline_output : nil
+      job_stats = output.pipeline_outputs.first ? output.pipeline_outputs.first.job_stats : nil
+      summary_stats = job_stats ? get_summary_stats(job_stats) : nil
+
+      output_data[:pipeline_info] = pipeline_info
+      output_data[:job_stats] = job_stats
+      output_data[:summary_stats] = summary_stats
+      final_result.push(output_data)
+    end
+    final_result
+  end
+
+  def samples_pipeline_run_info(samples)
+    pipeline_run_info = []
+    pipeline_run_entry = {}
+    samples.each do |output|
+      pipeline_run_status = output.pipeline_runs.first ? output.pipeline_runs.first.job_status : nil
+      pipeline_run_entry[:job_status_description] =
+        if %w[CHECKED SUCCEEDED].include?(pipeline_run_status)
+          'COMPLETE'
+        elsif %w[FAILED ERROR].include?(pipeline_run_status)
+          'FAILED'
+        elsif %w[RUNNING LOADED].include?(pipeline_run_status)
+          'IN PROGRESS'
+        else
+          'UPLOADING'
```

Comment:
I think you can add a ‘runnable’ and call it ‘initializing’

Reply:
okay, would "initializing" be a new status or replacing one of these @yunfangjuan?